### DISCUSSION TITLE
how to correctly transform 3D point cloud positions

### DISCUSSION THREAD

**Discussion by radoslavirha** [visgl/loaders.gl#2425] [Q&A]
Hi,

I'm trying to pick some points around cursor and highlight them and do some postprocessing in deck.gl.

Basically I'm trying to pick visually closest points to cursor with `pickObjects` which does not return WGS coordinates but points indexes.

I can find positions in `object.content.attributes.positions` and choose correct one by index, but I think it needs to be transformed/rotated. I'm facing similar issue than https://github.com/visgl/deck.gl/discussions/5714#discussioncomment-1940779.

So how can I

1. transform `position` correctly, which matrix from Tile3D should I use? Is the position only offset from cartesianOrigin?
2. then transform to WGS? I found `Ellipsoid.WGS84.cartesianToCartographic(transformed, new Vector3())`somewhere in loaders.gl code but when I render those point with PointClouLayer it's rotated as in ^^ comment

Thanks

**Comment by ibgreen** [comment]
Yes... without having time to look into details at this time, I suspect that having some transformPoint type methods on the Tile3D helper classes for working with individual points in the tiles coordinate system cold make a lot of sense. 

The matrices should all be there just having a methods that relieves the burden of figuring out how to use them from the user.

**Reply by radoslavirha** [reply]
Hi @ibgreen I created method which should return WGS coordinates of point by index. Can you help me with this? https://github.com/visgl/loaders.gl/pull/2498

Thanks

**Reply by radoslavirha** [reply]
Hi @ibgreen,

Sorry to bother you, have you had a chance to look at my PR?

Thanks
